app/api/entities/entities.js

```diff
@@ -56,42 +113,24 @@ function updateEntity(entity, _template) {
       if (typeof entity.template !== 'undefined') {
         d.template = entity.template;
       }
+
       return d;
     });
-
-    return Promise.all(docs.map(d => model.save(d)));
+    return Promise.all(docs.map(async _entity => inheritMetadata(_entity)))
+    .then(_docs => Promise.all(_docs.map(model.save)));
   });
 }
 
 function createEntity(doc, languages, sharedId) {
   const docs = languages.map((lang) => {
-    const langDoc = Object.assign({}, doc);
+    const langDoc = { ...doc, metadata: { ...doc.metadata } };
```

Comment:
Not sure how is this different from:
const langDoc = { ...doc};
Does it assign the metadata property (even if undefined) that it can be iterated later?

Reply:
When you do ` { ...doc};` you are cloning the object doc, but the property metadata points to the same metadata object as the original, so if you modify anything inside metadata you are also modifying the original.